src/main/java/org/opensearch/flowframework/workflow/ProcessNode.java

```diff
@@ -180,13 +182,10 @@ public CompletableFuture<WorkflowData> execute() {
                     delayExec.cancel();
                 }
                 logger.info("Finished {}.", this.id);
-            } catch (Throwable e) {
-                // TODO: better handling of getCause
-                this.future.completeExceptionally(e);
+            } catch (Throwable t) {
+                this.future.completeExceptionally(t.getCause() == null ? t : t.getCause());
             }
-            // TODO: improve use of thread pool beyond generic
-            // https://github.com/opensearch-project/flow-framework/issues/61
-        }, threadPool.generic());
+        }, threadPool.executor(PROVISION_THREAD_POOL));
```

Comment:
Should we change the name of the thread pool constant and string to make it more generic?
```
 public static final String WORKFLOW_THREAD_POOL = "opensearch_workflow";
```

Reply:
> Should we change the name of the thread pool constant and string to make it more generic?

Done.